common/json.c

```diff
@@ -250,17 +250,9 @@ void json_tok_print(const char *buffer, const jsmntok_t *tok)
 
 jsmntok_t *json_tok_copy(const tal_t *ctx, const jsmntok_t *tok)
 {
-	const jsmntok_t *first = tok;
-	const jsmntok_t *last = json_next(tok);
-
 	if (!tok)
 		return NULL;
-	jsmntok_t *arr = tal_arr(ctx, jsmntok_t, last - first);
-	jsmntok_t *dest = arr;
-	while (first != last)
-		*dest++ = *first++;
-
-	return arr;
+	return tal_dup_arr(ctx, jsmntok_t, tok, json_next(tok) - tok, 0);
```

Comment:
Maybe we should remove the NULL check and expect the caller to make sure `tok` is non-null.  This would only be needed in the call made by `json_check`.

Reply:
Good point: json_check() can never have NULL params, too.

My general rule is to give a nice crash on NULL unless there's a good reason to handle it.

Reply:
Since it's trivial, I'll fold and apply...